### DISCUSSION TITLE
Backblaze setup [Nov 08]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#4400] [Discord Threads]
> 🙋‍♂️ **3699n**

**Comment by o1lab** [comment]
> 🙋‍♂️ **wkw**  
#### 1. keyID = Access Key ID or Access Key   
#### Depending on the integration you’re using, you will be prompted for the Access Key ID or Access Key. In our S3 Compatibility API, this is the equivalent of our Application Key ID. You can see in the example here that this is shown at the top of that blue box.    
####    
#### 2. applicationKey = Secret Access Key or Secret Key   
#### Again, this will depend on the application that you’re using, but the Application Key is our equivalent of the Secret Key or Secret Access Key. In our example, this is shown at the bottom of the blue box.    
####    
#### 3. Region is the 2nd part of your S3 Endpoint   
#### Some integrations may need you to specify the region. The region will be the 2nd part of your S3 Endpoint. So if your S3 Endpoint is s3.us-west-001.backblazeb2.com, this means your region will be us-west-001.  
#  
> 🙋‍♂️ **wkw**  
#### but it seems the current version takes `s3.` as well  
#  
> 🙋‍♂️ **3699n**  
#### I receive an error "no such file or directory, open '/app/code/temp.txt"  
![image0](https://cdn.discordapp.com/attachments/1035046130863575050/1035173864365367398/unknown.png)  
#  
> 🙋‍♂️ **wkw**  
#### Click again  
#  
> 🙋‍♂️ **wkw**  
#### Is it latest version ?  
#  
> 🙋‍♂️ **wkw**  
#### We just released  
#  
> 🙋‍♂️ **3699n**  
#### it's 0.98.1  
#  
> 🙋‍♂️ **3699n**  
#### I keep receiving the same error when I press "test"  
#  
> 🙋‍♂️ **wkw**  
#### Try 0.98.2 . I fixed sth  
#  
> 🙋‍♂️ **3699n**  
#### Okay will do. Thank you  
#  
> 🙋‍♂️ **3699n**  
#### unfortunately I still receive the same error even on version 0.98.2  
![image0](https://cdn.discordapp.com/attachments/1035046130863575050/1035175245604196462/unknown.png)  
#  
> 🙋‍♂️ **3699n**  
#### ```   
#### Oct 27 12:56:24 errno: -2,   
#### Oct 27 12:56:24 code: 'ENOENT',   
#### Oct 27 12:56:24 syscall: 'open',   
#### Oct 27 12:56:24 path: '/app/code/temp.txt'   
#### Oct 27 12:56:24 }   
#### Oct 27 12:56:24 Error [Error: ENOENT: no such file or directory, open '/app/code/temp.txt'] {   
#### Oct 27 12:56:24 errno: -2,   
#### Oct 27 12:56:24 code: 'ENOENT',   
#### Oct 27 12:56:24 syscall: 'open',   
#### Oct 27 12:56:24 path: '/app/code/temp.txt'   
#### Oct 27 12:56:24 }   
#### Oct 27 12:56:24 p :: [Error: ENOENT: no such file or directory, open '/app/code/temp.txt'] {   
#### Oct 27 12:56:24 errno: -2,   
#### Oct 27 12:56:24 code: 'ENOENT',   
#### Oct 27 12:56:24 syscall: 'open',   
#### Oct 27 12:56:24 path: '/app/code/temp.txt'   
#### Oct 27 12:56:24 }   
#### Oct 27 12:56:25 POST /api/v1/db/meta/plugins/test 400 70 - 50.493 ms   
#### Oct 27 12:56:30 GET /dashboard 301 185 - 2.259 ms   
#### Oct 27 12:56:32 File Error [Error: ENOENT: no such file or directory, open '/app/code/temp.txt'] {   
#### Oct 27 12:56:32 errno: -2,   
#### Oct 27 12:56:32 code: 'ENOENT',   
#### Oct 27 12:56:32 syscall: 'open',   
#### Oct 27 12:56:32 path: '/app/code/temp.txt'   
#### Oct 27 12:56:32 }   
#### Oct 27 12:56:32 Error [Error: ENOENT: no such file or directory, open '/app/code/temp.txt'] {   
#### Oct 27 12:56:32 errno: -2,   
#### Oct 27 12:56:32 code: 'ENOENT',   
#### Oct 27 12:56:32 syscall: 'open',   
#### Oct 27 12:56:32 path: '/app/code/temp.txt'   
#### Oct 27 12:56:32 }   
#### Oct 27 12:56:32 p :: [Error: ENOENT: no such file or directory, open '/app/code/temp.txt'] {   
#### Oct 27 12:56:32 errno: -2,   
#### Oct 27 12:56:32 code: 'ENOENT',   
#### Oct 27 12:56:32 syscall: 'open',   
#### Oct 27 12:56:32 path: '/app/code/temp.txt'   
#### Oct 27 12:56:32 }   
#### ```  
#  
> 🙋‍♂️ **wkw**  
#### lemme check  
#  
> 🙋‍♂️ **wkw**  
#### looks fine on my side  
![image0](https://cdn.discordapp.com/attachments/1035046130863575050/1035185350974386206/unknown.png)  
#  
> 🙋‍♂️ **wkw**  
#### oh i get it sometimes when i keep pressing test  
![image0](https://cdn.discordapp.com/attachments/1035046130863575050/1035185473729085480/unknown.png)  
#  
> 🙋‍♂️ **wkw**  
#### maybe the temp file isnt generated before testing  
#  
> 🙋‍♂️ **3699n**  
#### Filed a bug report https://github.com/nocodb/nocodb/issues/4213  
#  
> 🙋‍♂️ **wkw**  
#### thanks will follow up  
#  
> 🙋‍♂️ **3699n**  
#### <@495466997304262658> Related question: If I import a project from Airtable when I have the B2 app enabled, will all the imported attachments be automatically uploaded and stored on B2?  
#  
> 🙋‍♂️ **wkw**  
#### lemme check with the team and get back to u later  
#  
> 🙋‍♂️ **3699n**  
#### Thanks!  
#  
> 🙋‍♂️ **3699n**  
#### Thanks for fixing the bug!  
#  
> 🙋‍♂️ **3699n**  
#### Just following up about my last question <@495466997304262658>  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1035046130863575050)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)

**Comment by YanickD** [comment]
I'm also having an issue trying to setup backblaze in nocodb, I get the following error when I hit the "Test" button:
![nocodbbackblaze-error](https://github.com/user-attachments/assets/fd98844c-4017-43b9-b8eb-d8dd3ad3463a)

Unsupported value for canned acl 'public-read'

**Reply by ssweber** [reply]
See https://github.com/nocodb/nocodb/discussions/10495

#### Access Control Lists (ACL) in B2
B2 bucket permissions ("Files in Bucket are:") must match your NocoDB configuration:

- In B2, set "Files in Bucket are:" to:
  - `Public` - Required for image previews (matches `public-read` NocoDB ACL field)
  - `Private` - More secure but disables previews (matches `private` NocoDB ACL field)
